### ISSUE TITLE
Azure storage block is currently not downloading flows to the correct relative path, results in FileNotFoundError

### ISSUE THREAD

**Issue by MrKriss** [PrefectHQ/prefect#9262]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

The Azure storage block is currently not downloading flows to the correct relative path, which results in a `FileNotFoundError` similar to the below, even though the files are uploaded correctly using the azure storage block. 

```
FileNotFoundError: [Errno 2] No such file or directory: '/opt/prefect/hello_flow.py'
```

The issue is likely due to upstream changes in `fsspec` as of versions `2023.3.0` and `2023.4.0`.  Downgrading to `2023.1.0` resolves the issue. However a current prefect install installs the latest breaking version `2023.4.0`.

### Reproduction

```python3
# Given the following local file layout
#
# ./to_upload/hello_flow.py
# ./to_upload/another_flow.py

from prefect.filesystems import Azure
import os

# Load storage and upload files
storage = Azure.load('my-storage')
storage.put_directory(local_path="./to_upload", to_path='my_files')

# Files on Azure
#
# ./my_files/hello_flow.py
# ./my_files/another_flow.py

# Attempt to download files to new directory
storage.get_directory(from_path='my_files', local_path="./new-dir")

# New Local Files
#
# ./new-dir/hello_flow.py
# ./new-dir/another_flow.py

# Attempt to download files to existing directory
os.mkdir('existing-dir')
storage.get_directory(from_path='my_files', local_path="./existing-dir")

# New Local Files
# Note extra directory level causes lookup of flow to fail 
#
# ./existing-dir/my_files/hello_flow.py  
# ./existing-dir/my_files/another_flow.py
```

### Error

```python3
Downloading flow code from storage at 'my_files'

Flow could not be retrieved from deployment.
Traceback (most recent call last):
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1016, in get_code
  File "<frozen importlib._bootstrap_external>", line 1073, in get_data
FileNotFoundError: [Errno 2] No such file or directory: '/opt/prefect/hello_flow.py'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/prefect/engine.py", line 276, in retrieve_flow_then_begin_flow_run
    flow = await load_flow_from_flow_run(flow_run, client=client)
  File "/usr/local/lib/python3.10/site-packages/prefect/client/utilities.py", line 40, in with_injected_client
    return await fn(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/prefect/deployments.py", line 217, in load_flow_from_flow_run
    flow = await run_sync_in_worker_thread(load_flow_from_entrypoint, str(import_path))
  File "/usr/local/lib/python3.10/site-packages/prefect/utilities/asyncutils.py", line 91, in run_sync_in_worker_thread
    return await anyio.to_thread.run_sync(
  File "/usr/local/lib/python3.10/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/usr/local/lib/python3.10/site-packages/prefect/flows.py", line 809, in load_flow_from_entrypoint
    flow = import_object(entrypoint)
  File "/usr/local/lib/python3.10/site-packages/prefect/utilities/importtools.py", line 201, in import_object
    module = load_script_as_module(script_path)
  File "/usr/local/lib/python3.10/site-packages/prefect/utilities/importtools.py", line 164, in load_script_as_module
    raise ScriptError(user_exc=exc, path=path) from exc
prefect.exceptions.ScriptError: Script at 'hello_flow.py' encountered an exception: FileNotFoundError(2, 'No such file or directory')
```

### Versions

```Text
❯ prefect version
Version:             2.10.4
API version:         0.8.4
Python version:      3.10.10
Git commit:          b6d0433a
Built:               Thu, Apr 13, 2023 5:34 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         cloud
```

### Additional context

This issue has been partly covered in https://github.com/PrefectHQ/prefect/issues/8710, relating to storage in s3, but the interim fix constrains `fsspec` in requirements.txt to not be version `2023.3.0`, and as the issue persists in `2023.4.0` it is now causing the latest prefect docker image to fail when using Azure Storage blocks.  

Additional testing revealed that including a trailing slash in the `from_path` seemed to give consistent behavior across 2023.1.0 (the last version of fsspec that did not have the issue) and 2023.4.0 (the current latest version). Results of testing included below:

```
fsspec version: 2023.1.0
from_path:  my_files
    New local directory
        ./new-dir/hello_flow.py
        ./new-dir/another_flow.py
    Existing local directory
        ./existing-dir/hello_flow.py
        ./existing-dir/another_flow.py
from_path:  my_files/
    New local directory
        ./new-dir/hello_flow.py
        ./new-dir/another_flow.py
    Existing local directory
        ./existing-dir/hello_flow.py
        ./existing-dir/another_flow.py

fsspec version: 2023.4.0
from_path:  my_files
    New local directory
        ./new-dir/hello_flow.py
        ./new-dir/another_flow.py
    Existing local directory
        ./existing-dir/my_files/hello_flow.py                  <-- Extra directory level causes lookup of flow to fail  
        ./existing-dir/my_files/another_flow.py             <-- Extra directory level causes lookup of flow to fail
from_path:  my_files/
    New local directory
        ./new-dir/hello_flow.py
        ./new-dir/another_flow.py
    Existing local directory
        ./existing-dir/hello_flow.py
        ./existing-dir/another_flow.py
```

**Comment by jmprovencher** [comment]
I have the same problem

**Comment by madkinsz** [comment]
Closed by #9440
